payjoin-cli/tests/e2e.rs

```diff
@@ -118,29 +118,28 @@ mod e2e {
 
             sigint(&cli_receiver).expect("Failed to kill payjoin-cli");
             sigint(&cli_sender).expect("Failed to kill payjoin-cli");
-            payjoin_sent
+            payjoin_sent.unwrap_or(Some(false)).expect("rx channel closed prematurely")
```

Comment:
Might it make more sense to handle this error on the line payjoin_sent is defined?

e.g. 

```rs
let payjoin_sent = tokio::time::timeout(timeout, rx.recv()).await
      .expect("timeout exceeded")  // Handle timeout error
      .expect("rx channel closed prematurely");  // Handle None from recv()
```